TECHMANIA/Assets/Scripts/Components/Main Menu/SelectTrackPanel.cs

```diff
@@ -659,8 +664,15 @@ public void OnRefreshButtonClick()
 
     public void OnGoUpButtonClick()
     {
-        currentLocation = new DirectoryInfo(currentLocation)
-            .Parent.FullName;
+        Debug.Log(currentLocation);
+
+        if (currentLocation.Contains(Paths.GetSaTrackRootFolder())) {
```

Comment:
Technically this will be incorrect when going up from subfolders inside streaming assets. How about this:

```
currentLocation = new DirectoryInfo(currentLocation).Parent.FullName;
if (currentLocation == Paths.GetSaTrackRootFolder()) currentLocation = Paths.GetTrackRootFolder();
```